client/src/main/scala/org/http4s/client/Http4sClientDsl.scala

```diff
@@ -30,13 +24,23 @@ trait Http4sClientDsl[F[_]] {
   }
 }
 
-object Http4sClientDsl {
+class MethodOps[F[_]](private val method: Method) extends AnyVal {
+
+  /** Make a [[org.http4s.Request]] using this [[Method]] */
+  final def apply(uri: Uri, headers: Header*)(implicit F: Applicative[F]): F[Request[F]] =
```

Comment:
Could you please say why the type of `F[Request[F]]` rather than just `Request[F]`? Thanks

Reply:
Uh, historical reasons?  `EntityEncoder` used to be effectful, which is why the overload is effectful, and this was effectful so the types didn't change when a body is added.  But that hasn't been true for a long time, and we'd like to eliminate the overloads in the client that take an effect of a request, at which case these would break.

I don't like this DSL anyway, but I think there's really no reason for it to continue to be effectful.